### PR TITLE
fix(rln-relay): RangeDefect test

### PR DESCRIPTION
**Pull request by rymnc**
This should resolve the broken ci tests for the epoch comparison in rln-relay.

Compares the epochs and returns the appropriate value.

### CODE REVIEW

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 1]
```diff
@@ -793,7 +793,10 @@ proc diff*(e1, e2: Epoch): int64 =
   let
     epoch1 = fromEpoch(e1)
     epoch2 = fromEpoch(e2)
-  return int64(epoch1) - int64(epoch2)
+  # Cannot use `abs` here because it is not available for `uint64`
+  if epoch1 > epoch2: return int64(epoch1 - epoch2)
+  else: return int64(epoch2 - epoch1) * -1
```
Comment (LNSD) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:798]:
```suggestion
  if epoch1 > epoch2: 
    return int64(epoch1 - epoch2)
  else: 	
    return int64(epoch2 - epoch1) * -1
```

**Review by LNSD — APPROVED** [review]
LGTM ✅

**Comment by LNSD** [comment]
> It doesn't really solve if epoch1 is uint64.high and epoch2 is uint64.low = 0: epoch1-epoch2 would have uint64 size and the cast to int64 (even if doesn't fail) will chop some bits. The only assumption that saves the test is that an epoch is 10s hence you reduce the epoch-width by few bits when converting from a timestamp. But the solution in general doesn't work (i.e. epoch = 1s)

Agree, it does not cover all the corner cases. See an example: https://play.nim-lang.org/#ix=4eAg

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim (hunk 1 above)
Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim]: Does it apply to the `safeIntCast` proc in [730e352](https://github.com/status-im/nwaku/pull/1318/commits/730e352b07edd2cea9e856f595965482b025025c) as well?

**Comment by rymnc** [comment]
> It doesn't really solve if epoch1 is uint64.high and epoch2 is uint64.low = 0: epoch1-epoch2 would have uint64 size and the cast to int64 (even if doesn't fail) will chop some bits. The only assumption that saves the test is that an epoch is 10s hence you reduce the epoch-width by few bits when converting from a timestamp. But the solution in general doesn't work (i.e. epoch = 1s)

This commit [730e352](https://github.com/status-im/nwaku/pull/1318/commits/730e352b07edd2cea9e856f595965482b025025c) ensures there is no overflow. wdyt?

**Comment by rymnc** [comment]
The only line we use `diff` on is - https://github.com/status-im/nwaku/blob/77465653415b1ca9cf58e2d0f69f9f09e36ae8a5/waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim#L829

and subsequently, we use the absolute value of the int64 returned, here - https://github.com/status-im/nwaku/blob/master/waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim#L828

A simple fix would be just to return the uint64 after checking which epoch is bigger, which is in [7746565](https://github.com/status-im/nwaku/pull/1318/commits/77465653415b1ca9cf58e2d0f69f9f09e36ae8a5)

**Review by staheri14 — APPROVED** [review]
Thanks, LGTM!

File: waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {.push raises: [Defect].}
 
 import
-  std/[sequtils, tables, times, streams, os, deques],
+  std/[sequtils, tables, times, streams, os, deques, bitops],
```
Comment (staheri14) [inline waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:4]: Where the `bitops` module is used?

Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim]: Addressed in [a42eb55](https://github.com/status-im/nwaku/pull/1318/commits/a42eb55c4a70df3d310471c58d810a4b262028cf), and removed `streams` as well.
